src/program/lwaftr/query/query.lua

```diff
@@ -87,17 +93,57 @@ local function print_counters (tree, filter)
    end
 end
 
+-- Return the pid that was specified, unless it was a leader process,
+-- in which case, return the follower pid that actually has useful counters.
+local function pid_to_parent(pid)
+   -- It's meaningless to get the parent of a nil 'pid'.
+   if not pid then return pid end
+   local pid = tonumber(pid)
+   for _, name in ipairs(shm.children("/")) do
+      local p = tonumber(name)
+      if p and ps.is_worker(p) then
+         local leader_pid = tonumber(ps.get_leader_pid(p))
+         -- If the precomputed by-name pid is the leader pid, set the pid
+         -- to be the follower's pid instead to get meaningful counters.
+      if leader_pid == pid then pid = p end
+      end
+   end
+   return pid
+end
+
 function run (raw_args)
    local opts, pid, counter_name = parse_args(raw_args)
-   if tostring(pid) and not counter_name then
-      counter_name, pid = pid, nil
+   if not opts.name then
+      if tostring(pid) then pid = pid_to_parent(pid) end
```

Comment:
I think `tostring` is not necessary. The reason is when pid is not a string? pid is not a string when pid is not defined (when the program is executed without args) So,

```lua
if pid then pid = pid_to_parent(pid) end
```

is equivalent.